ganttproject/src/net/sourceforge/ganttproject/chart/mouse/ChangeTaskProgressInteraction.java

```diff
@@ -61,15 +61,22 @@ private Task getTask() {
 
   @Override
   public void apply(MouseEvent event) {
-    int newProgress = myChangeScale.getProgress(event.getX());
+    ChangeTaskProgressRuler.Progress progress = myChangeScale.getProgress(event.getX());
+    int newProgress = progress.toPercents();
     if (newProgress > 100) {
       newProgress = 100;
     }
     if (newProgress < 0) {
       newProgress = 0;
     }
     myMutator.setCompletionPercentage(newProgress);
-    myLastNotes = new TaskInteractionHintRenderer(newProgress + "%", event.getX(), event.getY() - 30);
+    String hintText =
```

Comment:
You may like `MessageFormat` class or `String.format` method, and it probably makes sense to i18n it because every culture has its own understanding of proper spacing between numbers and percent signs (no kidding). I normally create a new key with formatting pattern like `{0} %({1}{2})` in `ganttproject/data/resources/language/i18n.properties` file. `GanttLanguage` is a handy singleton for accessing i18n keys and passing arguments.